src/ai/backend/common/types.py

```diff
@@ -1603,6 +1603,14 @@ def message(self) -> str:
         else:
             return "errors: " + "\n".join(self.errors)
 
+    @classmethod
+    def success(cls, result_type: ResultType) -> "DispatchResult[ResultType]":
+        return cls(result=result_type)
+
+    @classmethod
+    def error(cls, error_message: str) -> "DispatchResult[ResultType]":
```

Comment:
Why not use the Self type?

Reply:
mypy complains type error in below code, `Self type cannot have type arguments`.

```py
    @classmethod
    def success(cls, result_type: ResultType) -> Self[ResultType]:
        return cls(result=result_type)
```

Unfortunately, it seems that `Self` cannot be used together with generics.

Reply:
you can use `DispatchResult[ResultType]` without `""`